### PR TITLE
Fix container set kill

### PR DESCRIPTION
**Pull request by isubasinghe**
Fixes #TODO

Please do not open a pull request until you have checked ALL of these:

* [ ] Create the PR as draft .
* [ ] Run `make pre-commit -B` to fix codegen and lint problems.
* [ ] Sign-off your commits (otherwise the DCO check will fail).
* [ ] Use [a conventional commit message](https://www.conventionalcommits.org/en/v1.0.0/) (otherwise the commit message check will fail).
* [ ] "Fixes #" is in both the PR title (for release notes) and this description (to automatically link and close the issue).
* [ ] Add unit or e2e tests. Say how you tested your changes. If you changed the UI, attach screenshots.
* [ ] Github checks are green.
* [ ] Once required tests have passed, mark your PR "Ready for review".

If changes were requested, and you've made them, dismiss the review to get it reviewed again.

### CODE REVIEW

File: pkg/apis/workflow/v1alpha1/workflow_types.go [hunk 1]
```diff
@@ -1735,6 +1735,31 @@ func (s Nodes) Children(parentNodeId string) Nodes {
 	return childNodes
 }
 
+func (s Nodes) NestedChildrenStatus(parentNodeId string) ([]NodeStatus, error) {
```
Comment (JPZ13) [inline pkg/apis/workflow/v1alpha1/workflow_types.go:1740]: nit: Can you add a comment on this function explaining what it does?

**Comment by JPZ13** [comment]
@isubasinghe - for the failing tests, do you have any insights into what's going on? `Test_createSecretVolumesFromArtifactLocations_SSECUsed` seems to be passing on my local. I think the other test is a configuration thing. @tico24, can you take a quick look at the failing e2e test and chime in?

If we determine that the tests are due to flaky items or config, I'll approve and we can cut a release

**Comment by tico24** [comment]
I've already looked at the other thing. I got as far as going "this looks like code" and running away. Afraid I won't be much use here.

**Comment by JPZ13** [comment]
@tico24 - I tracked down when the mysql e2e test fails, and it looks like your CI changes broke it (not a bad thing, just a config thing). I think we can safely ignore that test

**Review by JPZ13 — APPROVED** [review]
Added one nit comment. I'll put out a PR to skip the e2e test that's breaking

**Comment by isubasinghe** [comment]
see here: https://github.com/pipekit/argo-workflows/pull/12

Moved because this branch was a shared PR.

**Pull request closed without merging**